C7Engine/AI/PlayerAI.cs

```diff
@@ -176,6 +180,39 @@ public static void SetAIForUnit(MapUnit unit, Player player)
 			}
 		}
 
+		public static bool UnitAttackingBarbCamp(MapUnit unit, Player player) {
+			if (unit.unitType.attack <= 0) {
+				return false;
+			}
+
+			List<Tile> reachableBarbCampsTiles = new List<Tile>();
+			foreach (Tile t in player.tileKnowledge.AllKnownTiles()
+				         .Where(t => unit.CanEnterTile(t, true) && t.cityAtTile == null && t.hasBarbarianCamp))
+			{
+				reachableBarbCampsTiles.Add(t);
+			}
```

Comment:
I see you've discovered LINQ but not fully on board yet ;)

`List<Tile> reachableBarbCampsTiles = player.tileKnowledge.AllKnownTiles().Where( ... ).toList()`

Reply:
Ah, I thought there ought to be a method like that but hadn't found it!  Switched!